[Wong Yew Jon] iP

wongyewjon opened the pull request:
##DukePro

>“What you lack in talent can be made up with desire, hustle, and giving 110% all the time” – Don Zimmer [source](https://sixads.net/blog/famous-quotes/).

DukePro frees your mind of having to remember things u need to do.
It's,

-text-based
-easy to learn
-~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/wongyewjon/ip/releases/download/v0.1/ipmain-all.jar)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :sunglasses:

And it is **FREE!**

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

This is the main method:
```
public class Main {
     public static void main(String[] args) {
         new Duke("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

yjonojy committed:
level 1

yjonojy committed:
changed Duke skeleton

yjonojy committed:
added a list class

yjonojy committed:
changes made

yjonojy committed:
task with ability to mark added

yjonojy committed:
Added Control, Todo, Deadline, Event

yjonojy committed:
test-ui-test

yjonojy committed:
edited some changes

yjonojy committed:
todo change

yjonojy committed:
no message

yjonojy committed:
text-ui-test

yjonojy committed:
added exceptions

yjonojy committed:
added delete  function

yjonojy committed:
Accept dates in format "dd/MM/yyyy" and print in a different format "MMM dd yyyy"

yjonojy committed:
Catches the case of wrong input of date format

yjonojy committed:
Accept deadline date format as "dd/mm/yyyy" and turn it into "mmm dd yyyy"

yjonojy committed:
oop in progress

yjonojy committed:
saving and loading of files in progress

yjonojy committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Control.java
#	src/main/java/Deadline.java

resolution: updated using the branch

yjonojy committed:
storage added

yjonojy committed:
storage fixed

yjonojy committed:
added into a package called duke

wongyewjon committed:
Merge remote-tracking branch 'remotes/upstream/add-gradle-support'

wongyewjon committed:
Used graddle

wongyewjon committed:
junit tests added

wongyewjon committed:
JavaDocs implemented

wongyewjon committed:
Complied with a coding standard

wongyewjon committed:
Code complied with a coding stardard

wongyewjon committed:
added find tasks using keyword

wongyewjon committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	build.gradle
#	src/main/java/duke/Duke.java
#	src/main/java/duke/EmptyDescriptionException.java
#	src/main/java/duke/InvalidDescriptionException.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Todo.java

wongyewjon committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

wongyewjon committed:
minor changes made

Eugenetanwl3881 commented on src/test/java/duke/TodoTest.java:
```diff
@@ -0,0 +1,23 @@
+public class TodoTest {
+    @Test
+    public void toStringTest() {
+        Todo temp1 = new Todo("", false);
```
Do you think there can be a better test variable name to use instead of _temp1_?

Eugenetanwl3881 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+    public void addTask(Task task) {
+        this.arrayList.add(task);
+        System.out.println(String.format("Got it. I've added this task:\n\t%s\n"
+                        + "Now you have %d task%s in the list.\n", task, this.arrayList.size(),
+                this.arrayList.size() == 1 ? "" : "s"));
+    }
+
+    /**
+     * Deletes an existing Task object from the array of Tasks.
+     * @param taskIndex The index of the Task to be deleted.
+     */
+    public void deleteTask(int taskIndex) {
+        Task temp = this.arrayList.remove(taskIndex - 1);
+        System.out.println(String.format(
+                "Noted. I've removed this task:\n\t%s\nNow you have %d task%s in the list.",
+                temp, this.arrayList.size(), this.arrayList.size() == 1 ? "" : "s"));
+    }
+
+    /**
+     * String representation of the list of tasks currently.
+     */
+    public void listTasks() {
+        System.out.println("Here are the tasks in your list:");
+
+        for (int i = 0; i < this.arrayList.size(); i++) {
+            Task currTask = this.arrayList.get(i);
+            System.out.println(String.format("    %d. %s", i + 1, currTask));
+        }
+    }
+
+    /**
+     * Finds the tasks given by a keyword inputted by the user.
+     * @param keyword Keyword used to search for tasks.
+     */
+    public void findKeyword(String keyword) {
+        ArrayList<Integer> tempArray = new ArrayList<>();
+        for (int i = 0; i < this.arrayList.size(); i++) {
+            if (this.arrayList.get(i).description.toLowerCase().contains(keyword.toLowerCase())) {
+                tempArray.add(i);
+            }
```
I like how your method names are clear by using verbs and written properly in camelCase. Good job! :smiley:

Eugenetanwl3881 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,197 @@
+     */
+    public void parseTodo(String[] subCmd) throws DukeException {
+
+        String tmp = String.join(" ", subCmd);
```
Do you think that a better naming for this variable can be _fullDescription_ instead of _tmp_ ?

Eugenetanwl3881 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,39 @@
+     *
+     * @return String representation of the Event.
+     */
+    @Override
```
Good that you remembered your Override annotations! It's easy to forget it sometimes 😊

Eugenetanwl3881 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+/**
+ * Encapsulates the main Duke program.
+ */
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
```
Clear use of names. Good job

Eugenetanwl3881 reviewed (commented):
Coding Standards PR review:
Your code follows good Java coding standards. Your code layout was good and the naming of methods and variables was generally good and easy to understand. Javadoc was also well documented. LGTM, just some small nits I pointed out in the comments. Keep up the good work!

wongyewjon committed:
Added GUI

yjonojy committed:
Added some assertions

yjonojy committed:
Parser classes: extract methods in Parser

Parser methods(e.g. ParseTodo, ParseDeadline) have some common
conditions to check (e.g. isEmptyString()).

The common conditions across the Parser methods cause code duplication.

Extracting the common conditions into a method allows for better
readability of code.

Let's pull up conditions common to methods into new methods named
isEmptyString() and isInValidIndex().

Extracting into single method reduces code duplication.

yjonojy committed:
Added CI

yjonojy committed:
changed CI

yjonojy committed:
reviewed checkstyle errors using CI

yjonojy committed:
changed checkstyle

yjonojy committed:
Create Enum PriorityLevel class

Tasks have different fixed priority levels.

Creating an Enum class with different levels of enumerations (e.g. LOW,
MEDIUM, HIGH, URGENT) as these levels are constants to be passed around.

Let's create an Enum class to allow easier representation of a group of
level constants.

wongyewjon committed:
Merge pull request #2 from wongyewjon/branch-C-Priority

Create Enum PriorityLevel class

yjonojy committed:
Merge branch 'master' into branch-A-Assertions

yjonojy committed:
Resolved merge conflicts

wongyewjon committed:
Merge pull request #3 from wongyewjon/branch-A-Assertions

Added some assertions

wongyewjon committed:
Merge branch 'master' into branch-A-CodeQuality

wongyewjon committed:
Merge pull request #4 from wongyewjon/branch-A-CodeQuality

Parser classes: extract methods in Parser

yjonojy committed:
Merge branch 'branch-A-Assertions'

yjonojy committed:
Merge branch 'master' of https://github.com/wongyewjon/ip

yjonojy committed:
Added priority schedule

yjonojy committed:
Merge branch 'branch-C-Priority'

yjonojy committed:
Improved GUI

yjonojy committed:
Adjusted dialogbox

yjonojy committed:
Improved Gui

yjonojy committed:
Added speech bubble

yjonojy committed:
Merge branch 'branch-A-BetterGui'

yjonojy committed:
Added case insensitive and trimmed messages functionality

yjonojy committed:
Reviewed Checkstyle

yjonojy committed:
Fixed elipses

yjonojy committed:
changed the text size and dialogueBox size

yjonojy committed:
Added User Guide

yjonojy committed:
Changed some User Guide

yjonojy committed:
Changed some Graddle

yjonojy committed:
Fixed checkstyle

yjonojy committed:
Editted Ui.png

yjonojy committed:
Fixed Gui bug

yjonojy committed:
Change Ui.png

yjonojy committed:
Edited User Guide

yjonojy committed:
Changed the UserGuide

yjonojy committed:
Updated User Guide
